src/effects/Repeat.cpp

```diff
@@ -141,25 +136,40 @@ bool EffectRepeat::Process(EffectInstance &, EffectSettings &)
          if (t0 > maxDestLen)
             maxDestLen = t0;
 
-         for (const auto pChannel : TrackList::Channels(&track)) {
-            auto clips = pChannel->SortedClipArray();
-            for (size_t i = 0; i < clips.size(); ++i) {
-               const auto eps = 0.5 / track.GetRate();
-               //Find first pasted clip
-               if (std::abs(clips[i]->GetPlayStartTime() - tc) > eps)
-                  continue;
-
-               //Fix pasted clips names
-               for(size_t j = 0; j < repeatCount; ++j) {
-                  for (size_t k = 0; k < clipNames.size(); ++k)
-                     if (i + k < clips.size())
-                        clips[i + k]->SetName(clipNames[k]);
-                  i += clipNames.size();
+         const auto compareIntervals = [](const auto& a, const auto& b) {
+            return a->Start() < b->Start();
+         };
+
+         const auto eps = 0.5 / track.GetRate();
+         auto sortedIntervals = std::vector(
+            track.Intervals().begin(),
+            track.Intervals().end()
+         );
+         auto sourceIntervals = std::vector(
+            firstTemp->Intervals().begin(),
+            firstTemp->Intervals().end()
+         );
+         std::sort(sortedIntervals.begin(), sortedIntervals.end(), compareIntervals);
+         std::sort(sourceIntervals.begin(), sourceIntervals.end(), compareIntervals);
```

Comment:
This getting of a sorted sequence of intervals looks like something that might be frequently needed and could become a utility function.  If in `lib-channel`, it might need a template type parameter for more convenience.

Do that another day...